pool/balancing.go

```diff
@@ -0,0 +1,14 @@
+package pool
+
+import "github.com/tarantool/go-tarantool/v2"
+
+type BalancingMethod = func(int) BalancingPool
```

Comment:
I prefer an interface instead of a method. An interface could be used as a method, but doing the opposite is more difficult. But this is debatable. 
```Go
type BalancerFactory interface {
    Create() Balancer
}
```
As additional we could pass some useful info into the method:
```Go
type BalancerFactory interface {
    Create(role Role) Balancer
}
```
Or we can keep it simple.